src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge29.java

```diff
@@ -0,0 +1,89 @@
+package org.owasp.wrongsecrets.challenges.docker;
+import org.owasp.wrongsecrets.RuntimeEnvironment;
+import org.owasp.wrongsecrets.ScoreCard;
+import org.owasp.wrongsecrets.challenges.*;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.SecureRandom;
+import java.util.List;
+//import java.net.HttpURLConnection;
+//import java.net.URL;
+import java.util.Random;
+//import java.util.Scanner;
+
+@Component
+@Order(29)
+
+
+public class Challenge29 extends Challenge {
+    private final Random secureRandom = new SecureRandom();
+    private static final String alphabet = "0123456789QWERTYUIOPASDFGHJKLZXCVBNMqwertyuiopasdfghjklzxcvbnm";
+    public String sol = getMyString();
```

Comment:
it's easier if we just use a get-er instead of this variable :) can you remove this and instead where requested return `challenge.spoiler().solution() ` please? that way we don't add more relationships between objects that will make it harder to debug.